File: solutions/python/series/1/series.py
def slices(series, length):
    # Error: empty series
    if series == "":
        raise ValueError("series cannot be empty")

    # Error: slice length zero
    if length == 0:
        raise ValueError("slice length cannot be zero")

    # Error: negative slice length
    if length < 0:
        raise ValueError("slice length cannot be negative")

    # Error: slice length longer than series
    if length > len(series):
        raise ValueError("slice length cannot be greater than series length")

    # Generate all contiguous slices
    return [series[i:i + length] for i in range(len(series) - length + 1)]

File: solutions/java/series/1/src/main/java/Series.java
import java.util.ArrayList;
import java.util.List;

class Series {

    private final String digits;

    Series(String digits) {
        if (digits.isEmpty()) {
            throw new IllegalArgumentException("series cannot be empty");
        }
        this.digits = digits;
    }

    List<String> slices(int n) {
        if (n <= 0) {
            throw new IllegalArgumentException("slice length cannot be negative or zero");
        }
        if (n > digits.length()) {
            throw new IllegalArgumentException("slice length cannot be greater than series length");
        }

        List<String> result = new ArrayList<>();
        for (int i = 0; i <= digits.length() - n; i++) {
            result.add(digits.substring(i, i + n));
        }
        return result;
    }
}